tests/unit/test-cmd/test.sign.js

```diff
@@ -278,8 +278,10 @@ describe('sign', () => {
                        stubs.signingConfig.timeout);
           assert.equal(signedAddonCall.xpiPath,
                        stubs.buildResult.extensionPath);
-          assert.equal(signedAddonCall.id,
-                       stubs.preValidatedManifest.applications.gecko.id);
+
+          const applications = stubs.preValidatedManifest.applications || {};
```

Comment:
@kumar303

nit, it is pretty weird that if we omit `|| {}` here, then flow complains that we are trying to access `gecko` on a "possibly undefined value", but it doesn't complain that we are trying to access `id` on the "possibly undefined `gecko` property".

I checked which is the type detected by flow using `flow type-at-pos` on `applications` from line 283 and it is inferenced as the object defined in the manifest helper module:
```
{gecko: {id?: string, strict_min_version?: string, strict_max_version?: string, update_url?: string}}
```

on the contrary, if we define the type explicitly and use the type annotation on line 282, flow is complaining with {} not being of the valid type.

Here is the tweaked version that make this type check stronger:

src/utils/manifest.js
```
export type ExtensionManifestApplications = {|
  gecko: {|
    id?: string,
    strict_min_version?: string,
    strict_max_version?: string,
    update_url?: string,
  |},
|};

export type ExtensionManifest = {
  name: string,
  version: string,
  default_locale?: string,
  applications?: ExtensionManifestApplications,
};
```

test/unit/cmd/test.sign.js:
```
const applications: ExtensionManifestApplications =
    stubs.preValidatedManifest.applications || {gecko: {}};
assert.equal(signedAddonCall.id, applications.gecko.id);
```